build/Dockerfile

```diff
@@ -6,7 +6,7 @@ USER 0
 RUN bash -c -- 'if [ "${NO_PROXY:-__ZZZZZ}" == "__ZZZZZ" ]; then echo "Applying upstream EPEL hacks" && echo -e "-----BEGIN PGP PUBLIC KEY BLOCK-----\n\nmQINBFz3zvsBEADJOIIWllGudxnpvJnkxQz2CtoWI7godVnoclrdl83kVjqSQp+2\ndgxuG5mUiADUfYHaRQzxKw8efuQnwxzU9kZ70ngCxtmbQWGmUmfSThiapOz00018\n+eo5MFabd2vdiGo1y+51m2sRDpN8qdCaqXko65cyMuLXrojJHIuvRA/x7iqOrRfy\na8x3OxC4PEgl5pgDnP8pVK0lLYncDEQCN76D9ubhZQWhISF/zJI+e806V71hzfyL\n/Mt3mQm/li+lRKU25Usk9dWaf4NH/wZHMIPAkVJ4uD4H/uS49wqWnyiTYGT7hUbi\necF7crhLCmlRzvJR8mkRP6/4T/F3tNDPWZeDNEDVFUkTFHNU6/h2+O398MNY/fOh\nyKaNK3nnE0g6QJ1dOH31lXHARlpFOtWt3VmZU0JnWLeYdvap4Eff9qTWZJhI7Cq0\nWm8DgLUpXgNlkmquvE7P2W5EAr2E5AqKQoDbfw/GiWdRvHWKeNGMRLnGI3QuoX3U\npAlXD7v13VdZxNydvpeypbf/AfRyrHRKhkUj3cU1pYkM3DNZE77C5JUe6/0nxbt4\nETUZBTgLgYJGP8c7PbkVnO6I/KgL1jw+7MW6Az8Ox+RXZLyGMVmbW/TMc8haJfKL\nMoUo3TVk8nPiUhoOC0/kI7j9ilFrBxBU5dUtF4ITAWc8xnG6jJs/IsvRpQARAQAB\ntChGZWRvcmEgRVBFTCAoOCkgPGVwZWxAZmVkb3JhcHJvamVjdC5vcmc+iQI4BBMB\nAgAiBQJc9877AhsPBgsJCAcDAgYVCAIJCgsEFgIDAQIeAQIXgAAKCRAh6kWrL4bW\noWagD/4xnLWws34GByVDQkjprk0fX7Iyhpm/U7BsIHKspHLL+Y46vAAGY/9vMvdE\n0fcr9Ek2Zp7zE1RWmSCzzzUgTG6BFoTG1H4Fho/7Z8BXK/jybowXSZfqXnTOfhSF\nalwDdwlSJvfYNV9MbyvbxN8qZRU1z7PEWZrIzFDDToFRk0R71zHpnPTNIJ5/YXTw\nNqU9OxII8hMQj4ufF11040AJQZ7br3rzerlyBOB+Jd1zSPVrAPpeMyJppWFHSDAI\nWK6x+am13VIInXtqB/Cz4GBHLFK5d2/IYspVw47Solj8jiFEtnAq6+1Aq5WH3iB4\nbE2e6z00DSF93frwOyWN7WmPIoc2QsNRJhgfJC+isGQAwwq8xAbHEBeuyMG8GZjz\nxohg0H4bOSEujVLTjH1xbAG4DnhWO/1VXLX+LXELycO8ZQTcjj/4AQKuo4wvMPrv\n9A169oETG+VwQlNd74VBPGCvhnzwGXNbTK/KH1+WRH0YSb+41flB3NKhMSU6dGI0\nSGtIxDSHhVVNmx2/6XiT9U/znrZsG5Kw8nIbbFz+9MGUUWgJMsd1Zl9R8gz7V9fp\nn7L7y5LhJ8HOCMsY/Z7/7HUs+t/A1MI4g7Q5g5UuSZdgi0zxukiWuCkLeAiAP4y7\nzKK4OjJ644NDcWCHa36znwVmkz3ixL8Q0auR15Oqq2BjR/fyog==\n=84m8\n-----END PGP PUBLIC KEY BLOCK-----" > /etc/pki/rpm-gpg/RPM-GPG-KEY-EPEL-8 && echo -e "[epel]\nname=Extra Packages for Enterprise Linux 8 - \$basearch\nmetalink=https://mirrors.fedoraproject.org/metalink?repo=epel-8&arch=\$basearch&infra=\$infra&content=\$contentdir\nenabled=1\ngpgcheck=1\ngpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-EPEL-8" > /etc/yum.repos.d/epel.repo; fi'
 
 # Required for oauth-proxy
-RUN dnf install -y python3-passlib python3-bcrypt
+RUN dnf install -y python3-passlib python3-bcrypt && dnf update -y --exclude=ansible*
```

Comment:
Well this is upstream, which means I'm a bit bullish to have everything as up to date as possible, which can help identify issues earlier one. Since this doesn't ship downstream, I'm a bit less concerned about image size. You're right that cleaning up the cache will certainly result in a smaller image size.

I'm working with the ansible-operator team to try and get their images to cachito integration which should make updates significantly simpler, but I'm waiting on ART to do an initial import so I can verify my changes are working. I don't expect to see this happen until November timeframe.

The idea around installing via pip is basically what should happen once moving to cachito integration so that alignment with RPMs and such isn't as critical.

I'm also ok with us doing nothing here and relying on the image as-is. I just happened to be looking at the image scans in quay.io and noticed that a simple `dnf update` would result in them going green.

Reply:
```suggestion
RUN dnf update -y --exclude=ansible* && dnf install -y python3-passlib python3-bcrypt && dnf clean all && rm -rf /var/cache/dnf


```

Reply:
Joining a bit late to this convo. Agree with Emma that we should rely on the origin-ansible-operator image being updated. IMHO we should follow up with them and share our findings (I don't mind chasing this if you agree). If we want to make sure we are testing with the latest available and that is something that doesn't adjust ansible-operator team roadmap, I'm ok with the idea of adding this to our Dockerfiles.

I'm not very convinced with the idea of installing from pip. I think we should stick as closely as possible with what we are shipping.

Lastly, please add some docs (just inline comments is good) to reflect what is going on with this line.

Reply:
I'm working with the team responsible for helping make the image building simpler (via cachito) and I'm hoping the eventual result of that is a better base image. My opinion is that we should update the image upstream for a couple reasons outlined:

1) optimistic forward-looking testing to try and catch any regressions early
2) helps pass security scans in our upstream artifacts
3) does not affect downstream images (separate Dockerfile template for that)
4) our downstream images also run a `dnf update` :)

Our downstream image runs this (which looks to also require an update for efficiency and image size reasons as identified here):

```
# mitigate CVE issues by updating base image
USER root
RUN yum update -y
RUN yum install -y python3-passlib python3-bcrypt
```